core-bundle/src/Event/FrontendMenuEvent.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Event;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class FrontendMenuEvent extends Event
+{
+    private FactoryInterface $factory;
+    private ItemInterface $tree;
+    private int $pid;
+    private int $level;
+    private array $options;
+
+    public function __construct(FactoryInterface $factory, ItemInterface $tree, int $pid, int $level, array $options)
```

Comment:
I am not sure about the `$options` array. For example - if I want to change the menu _only_ for the `navigation` module, but not for the `customnav` module, I could check for `'navigation' === $event->getOptions()['type']` - however, since the options can contain data from basically anything, I can never be sure that this was actually a Contao module of type `navigation`.

Reply:
We can make some options mandatory, but I don't think any need for action in that concern.
What I currently don't like on the $options array is that we have an option called 'isSitemap' that is not something coming from the module row.

Reply:
I rather meant that the event should may be store the module it originated from so that you could check the module type. Though I am not sure that's a good idea either.

Reply:
The MenuBuilder should not have a dependency on Modules IMHO.